DESCW-3209 updates all npm packages as far as safely possible without…

robrien-bcgov opened the pull request:
## test this here https://notify-bc-c345c5-dev.apps.silver.devops.gov.bc.ca/configurations/

### Note: Issue https://github.com/bcgov/des-notifybc/issues/132 prevents upgrading deps in the /client for now. I have tried updating parts of the dependency chain (vue/vuetify/vite) and their transitive deps, but this either breaks the build locally, or triggers a known bug in NPM that removes optional dependencies (in this case, the linux version of the `rollout` package. 
see this log for more info on that: https://github.com/bcgov/des-notifybc/actions/runs/20082134520/job/57611696447

---

This pull request updates the project's dependencies and devDependencies in `package.json` to their latest compatible versions. These upgrades include core NestJS packages, various utility libraries, and development tools, which can provide bug fixes, security patches, and new features.

Dependency updates:

* Upgraded core dependencies such as `@nestjs/common`, `@nestjs/core`, `@nestjs/mongoose`, `@nestjs/swagger`, and `@nestjs/terminus` to their latest 10.x versions for improved compatibility and features.
* Updated widely-used libraries including `async`, `axios`, `helmet`, `ioredis`, `jsonwebtoken`, `mailparser`, `mongoose`, `nodemailer`, `rxjs`, `semver`, `smtp-server`, and `twilio` to their latest minor or patch versions, addressing potential bugs and security issues.
* Changed the `jmespath` dependency to use the GitHub repository reference, ensuring the correct version source.

Development tooling updates:

* Upgraded devDependencies such as `@nestjs/cli`, `@nestjs/schematics`, `@nestjs/testing`, TypeScript type definitions, ESLint, Prettier, `ts-jest`, `ts-node`, and `typescript` for better development experience and compatibility.
* Updated `redis-memory-server` in `optionalDependencies` to

robrien-bcgov committed:
DESCW-3209 updates all npm packages as far as safely possible without breaking changes

- build still works
- may need confirmation later

ShawnTurple reviewed (commented):
All this looks good, however we can't merge just yet, until we deploy this to test, but currently @mhaswell-bcgov is working on some configuration changes to facilitate this testing, so please move this ticket to blocked for now.

robrien-bcgov committed:
descw-3209 update package versions using Node v24

robrien-bcgov committed:
descw-3209 config was importing a package wrong - fixed

robrien-bcgov committed:
descw-3209 start troubleshooting build with node v20 and no updates to npm deps

robrien-bcgov committed:
desw-3209 revert changes to ./client deps

robrien-bcgov committed:
enable production source maps for debugging vite/react code in client page table

robrien-bcgov committed:
descw-3209 revert most changes

robrien-bcgov committed:
descw-3209 update jquery in client to trigger build tag hopefully

robrien-bcgov committed:
descw-3209 maximum updates for root deps

robrien-bcgov committed:
descw-3209 minimal changes to client deps

robrien-bcgov committed:
descw-3209 fix import shape in vite config for client

robrien-bcgov committed:
descw-3209 undo minimatch override

robrien-bcgov committed:
descw-3209 hopefully this builds a working client image

robrien-bcgov committed:
descw-3209 revert deps changes

robrien-bcgov committed:
full revert of client deps

robrien-bcgov committed:
descw-3209 manually upgrade vite & vuetify in build deps for client

- this required a change to the named imports in the vite config
- this also required a change to the named imports in the vuetify config

robrien-bcgov committed:
attempt to force the linux x64 optional rollup package to be installed

robrien-bcgov committed:
allow build to use the optional packages so the x64 linux rollup can be installed

robrien-bcgov committed:
revert dockerfile and client changes for stability

- the vue/vite/vuetify ecosystem is too fragile for these version updates.
- the npm bug kept stripping the linux x64 flavor of rollup and killing the last build stage
- I am tempted to keep the Dockerfile changes, but why tempt fate?

mhaswell-bcgov reviewed (approved):
It's up on test right now and seems to be working well, so this looks good to go.
